### DISCUSSION TITLE
Login with Cognito: always throwing an INVALID_USER error

### DISCUSSION THREAD

**Discussion by HakimJu** [directus/directus#11809] [Archive]
I cannot login using Cognito when clicking on "Log In with Cognito" button. 
I made sure in the database that the user's provider is "cognito" and  regarding "External Identifier" its value is the user's email address I also tried to use Cognito's sub ID but both failed in my case.

Here's the env variables:

```yaml
AUTH_PROVIDERS: cognito
AUTH_COGNITO_DRIVER: oauth2
AUTH_COGNITO_CLIENT_ID: <clientId>
AUTH_COGNITO_CLIENT_SECRET: <clientSecret>
AUTH_COGNITO_AUTHORIZE_URL: https://abc-auth.auth.eu-central-1.amazoncognito.com/oauth2/authorize
AUTH_COGNITO_ACCESS_URL: https://abc-auth.auth.eu-central-1.amazoncognito.com/oauth2/token
AUTH_COGNITO_PROFILE_URL: https://abc-auth.auth.eu-central-1.amazoncognito.com/oauth2/userInfo
#AUTH_COGNITO_ALLOW_PUBLIC_REGISTRATION: "true"
#AUTH_COGNITO_DEFAULT_ROLE_ID: user-role-id
AUTH_COGNITO_SCOPE: "openid email profile"
AUTH_COGNITO_ICON: aws
#REFRESH_TOKEN_COOKIE_DOMAIN: https://abc-auth.auth.eu-central-1.amazoncognito.com
REFRESH_TOKEN_COOKIE_SECURE: "true"
REFRESH_TOKEN_COOKIE_SAME_SITE: none
REFRESH_TOKEN_COOKIE_NAME: abc_refresh_token
```

### To Reproduce

1. Click on "Log In with Cognito" button

2. Correctly set "Username" + "Password" fields

3. Click on "Sign In" button

4. On success, redirected to /admin/login?reason=INVALID_USER

<details>
<summary>More info</summary>

### Errors Shown

Invalid User

### What version of Directus are you using?

9.5

### What version of Node.js are you using?

16.11.0

### What database are you using?

Postgres 13

### What browser are you using?

Chrome

### What operating system are you using?

macOS

### How are you deploying Directus?

Docker

</details>

**Comment by aidenfoxx** [comment]
The library we use for oauth/openid will reject openid requests under oauth2 protocol. You need to either remove "openid" from your scope, or actually use the openid driver. Preferably the latter.

**Reply by HakimJu** [reply]
Very good hint! Thank you! I switched to use the openid driver and it worked like charm
